pkg/dataflatten/json.go

```diff
@@ -4,13 +4,26 @@ import (
 	"encoding/json"
 	"fmt"
 	"os"
+
+	"golang.org/x/text/encoding/unicode"
+	"golang.org/x/text/transform"
 )
 
 func JsonFile(file string, opts ...FlattenOpts) ([]Row, error) {
 	rawdata, err := os.ReadFile(file)
 	if err != nil {
 		return nil, err
 	}
+
+	// Attempt to decode utf16 data.
+	valid_json := json.Valid(rawdata)
+	if !valid_json {
+		rawdata, _, err = transform.Bytes(unicode.UTF16(unicode.LittleEndian, unicode.UseBOM).NewDecoder(), rawdata)
```

Comment:
Is there a way to detect this, or is just trying it the simplest pattern?

Reply:
Best I could find at the moment is https://pkg.go.dev/github.com/saintfish/chardet, but I didn't dig very deep yet since there's not much info on it.

Reply:
`json.Valid` seems reasonable. Pulling in another library is always mixed.